sync/sync.go

```diff
@@ -322,7 +322,11 @@ Register:
 	}
 	if action == "start" {
 		logrus.Infof("Received start from controller")
-		return t.client.Start(replicaAddress)
+		if err := t.client.Start(replicaAddress); err != nil {
+			return err
+		}
+		go s.Preload()
```

Comment:
Care also needs to be taken while calling s.Preload() without a lock. If s.r is closed while preload is going on, replica might show unexpected behavior.